UDV-Core/src/Utils/BaseDemo/js/BaseDemo.js

```diff
@@ -479,18 +485,165 @@ export class BaseDemo {
+                                    result.traverse(
+                                        function _setPointSize(mesh) {
+                                            if (mesh.material) {
+                                                mesh.material.size = size;
+                                            }
+                                        });
+                                }
+                            }
+                            ;
+                            config.update = itowns.FeatureProcessing.update;
+                            config.convert = itowns.Feature2Mesh.convert({
+                                color: new itowns.THREE.Color().setRGB(
+                                    config.color.red,
+                                    config.color.green,
+                                    config.color.blue
+                                ),
+                                altitude: config.altitude,
+                            });
+                            config.overrideAltitudeInToZero = true;
+                            config.projection = config.source.projection;
+                            config.source = config.source;
+                            let layer = new itowns.GeometryLayer(
+                                config.id,
+                                new itowns.THREE.Group(),
+                                config
+                            );
+                            return this.view.addLayer(layer);
+                        };
+
+                        //// If no protocol, set configuration
+                        if (protocol === undefined) {
+                            let configuration = configList[i];
+                            var promise = new Promise(
+                                function (resolve, reject) {
+                                    configuration.source =
+                                        new itowns.FileSource(
+                                            configList[i].source);
+                                    resolve(configuration);
+                                }
+                            );
+                            promise.then(addGeometryLayerFromConfig);
+                        }
+
+                        //// If protocol === "wfs", set configuration
+                        else if (protocol === "wfs") {
+                            let configuration = configList[i];
+                            var promise = new Promise(
+                                function (resolve, reject) {
+                                    configuration.source = new itowns.WFSSource(
+                                        configList[i].source);
+                                    resolve(configuration);
+                                }
+                            );
+                            promise.then(addGeometryLayerFromConfig);
+                        }
+                    }
+                }
+            );
```

Comment:
The code you added in this method should be in a separate method of BaseDemo (e.g. `addVectorLayers()`) that takes a configuration file as argument. This way, only the users that want these layers in their demo can add them with baseDemo.addVectorLayers(configFile).